test/test-clusters/src/main/java/org/elasticsearch/test/cluster/local/AbstractLocalClusterFactory.java

```diff
@@ -912,15 +917,39 @@ private Map<String, String> getEnvironmentVariables() {
             return environment;
         }
 
-        private Map<String, String> getJvmOptionsReplacements() {
-            return Map.of(
-                "-XX:HeapDumpPath=data",
-                "-XX:HeapDumpPath=" + logsDir,
-                "logs/gc.log",
-                logsDir.resolve("gc.log").toString(),
-                "-XX:ErrorFile=logs/hs_err_pid%p.log",
-                "-XX:ErrorFile=" + logsDir.resolve("hs_err_pid%p.log")
-            );
+        private record ReplacementKey(String key, String fallback) {}
+
+        private Map<ReplacementKey, String> getJvmOptionsReplacements() {
```

Comment:
This become much more complicated, mirroring the ElasticsearchNode logic. Is this intentional? (And the question about ifs on version still stands: I don't understand why we need them)

Reply:
My point is: the previous code was much simpler, and now it contains all the logic for launching 7.x and even 6.x clusters -- do we need it?

Reply:
I don't think the previous code was exactly simpler. There was a discrepancy between this code (the new test infra) and the old code (ElasticsearchNode). The changes to this function bring them inline with each other. We do need it because at minimum there is a difference in the jvm.options file between 8.19+ and <=8.18.x. The handling of before 6.3 is done in the else cases. I'm happy to remove that, but I would like to do it in a followup so we can be sure no tests are actually relying on it, consistently across old and new test infrastructure.